src/api/utils/BuildQuery.ts

```diff
@@ -0,0 +1,75 @@
+import {
+  DefaultFilter,
+  Filter,
+  HasFilter,
+  SearchFilter,
+} from "../filters/Filter";
+
+enum Operator {
+  eq = "=",
+  gt = ">",
+  ls = "<",
+  has = "has",
+  search = "search",
+}
+
+function buildPaginationQuery(limit: number, offset: number) {
+  return ` LIMIT ${limit} OFFSET ${offset};`;
+}
+
+function buildFilterQuery(filters: Filter[], alias: string) {
+  if (!filters || filters.length === 0 || !Array.isArray(filters)) {
+    return "";
+  }
```

Comment:
```suggestion
  if (!filters || !Array.isArray(filters)) {
    return "";
  }
```
Redundância nos checks, a não ser que filters possa ser uma string também